platform/features/timeline/src/controllers/swimlane/TimelineSwimlaneDropHandler.js

```diff
@@ -129,17 +155,12 @@ define(
                 });
             }
 
-            // Check if a compose action is allowed for the object in this
-            // swimlane (we handle the link differently to set the index,
-            // but check for the existence of the action to invole the
-            // relevant policies.)
-            function allowsCompose(swimlane, domainObject) {
-                var actionCapability =
-                    swimlane.domainObject.getCapability('action');
-                return actionCapability && actionCapability.getActions({
-                    key: 'compose',
-                    selectedObject: domainObject
-                }).length > 0;
+            function drop(domainObject, targetObject, indexOffset) {
+                var action = chooseAction(targetObject, domainObject);
+                return action.perform().then(function () {
```

Comment:
I see, makes sense. Might be clearer if all of the logic were in the one place. eg

``` javascript
function isParentOf(domainObject, potentialParent){
    var context = domainObject.getCapability('context'),
          parent = context && context.getParent(),
          parentId = parent && parent.getId();
          return parentId === potentialParent.getId();
}
...
    allowDropIn: function (id, domainObject) {
                     return inEditMode() &&
                     !pathContains(swimlane, id) &&
                     !contains(swimlane, id) &&
                     (chooseAction(swimlane.domainObject, domainObject)!=='undefined' || 
                     isParentOf(swimland.domainObject, domainObject));
                 },
```

The logic returned by that function is long-winded, but it at least states everything in one place.